packages/cryptography/src/Mnemonic.js

```diff
@@ -125,7 +144,61 @@ export default class Mnemonic {
             return this.toLegacyPrivateKey();
         }
 
-        return this._toPrivateKey(passphrase);
+        const seed = await bip39.toSeed(this.words, passphrase);
+        const digest = await hmac.hash(hmac.HashAlgorithm.Sha512, "ed25519 seed", seed);
+
+        let keyData = digest.subarray(0, 32);
+        let chainCode = digest.subarray(32);
+
+        if (CACHE.privateKeyConstructor == null) {
+            throw new Error("PrivateKey not found in cache");
+        }
+
+        for (const index of path) {
+            ({ keyData, chainCode } = await slip10.derive(
+                keyData,
+                chainCode,
+                index
+            ));
+        }
+
+        const keyPair = nacl.sign.keyPair.fromSeed(keyData);
+
+        return CACHE.privateKeyConstructor(
+            new Ed25519PrivateKey(keyPair, chainCode)
+        );
+    }
+
+    /**
+     * Recover an ECDSA private key from this mnemonic phrase, with an
+     * optional passphrase.
+     *
+     * @param {string} [passphrase]
+     * @param {number[]} [path]
+     * @returns {Promise<PrivateKey>}
+     */
+    async toEcdsaPrivateKey(passphrase = "", path = ECDSA_ETH_PATH) {
+        const seed = await bip39.toSeed(this.words, passphrase);
+        const digest = await hmac.hash(hmac.HashAlgorithm.Sha512, "Bitcoin seed", seed);
+
+        let keyData = digest.subarray(0, 32);
```

Comment:
`const seed` up to `let chainCode` is mostly dup code.
Could pull out into a shared method that take in `passphrase` and `seedText` and returns `keyData` and `chainCode`